chore: documenting simplified versioning (#658)
snyk/sweater-comb: docs/principles/version.md

Versions for the entire API are expressed in the form _`{version date}~{stability level}`_, assuming GA was not specified.

## Introducing Simplified Versioning (as of 2024-10-15)

> As of **2024-10-15**, we have introduced a "simplified versioning" model to further streamline API version management. This change introduces a "synthetic" version, which impacts both the developer and consumer sides of the API in a few important ways:

Reviewer: nit: use of the first person in doc

```suggestion
As of **2024-10-15**,  "simplified versioning" model was introduced to further streamline API version management. This change introduces a "synthetic" version, which impacts both the developer and consumer sides of the API in a few important ways:
```
